src/sky.cpp

```diff
@@ -613,56 +787,73 @@ void Sky::update(float time_of_day, float time_brightness,
 	float color_change_fraction = 0.98;
 	if (sunlight_seen) {
 		if (is_dawn) {  // Dawn
-			m_bgcolor_bright_f = m_bgcolor_bright_f.getInterpolated(
-				bgcolor_bright_dawn_f, color_change_fraction);
-			m_skycolor_bright_f = m_skycolor_bright_f.getInterpolated(
-				skycolor_bright_dawn_f, color_change_fraction);
+			m_sky_color_bottom_bright_f = m_sky_color_bottom_bright_f.getInterpolated(
+				sky_bottom_bright_dawn_f, color_change_fraction);
+
+			m_sky_color_top_bright_f = m_sky_color_top_bright_f.getInterpolated(
+				sky_top_bright_dawn_f, color_change_fraction);
+
 			m_cloudcolor_bright_f = m_cloudcolor_bright_f.getInterpolated(
 				cloudcolor_bright_dawn_f, color_change_fraction);
 		} else {
-			if (time_brightness < 0.07) {  // Night
-				m_bgcolor_bright_f = m_bgcolor_bright_f.getInterpolated(
-					bgcolor_bright_night_f, color_change_fraction);
-				m_skycolor_bright_f = m_skycolor_bright_f.getInterpolated(
-					skycolor_bright_night_f, color_change_fraction);
+			if (time_brightness < 0.08) {  // Night
+				m_sky_color_bottom_bright_f = m_sky_color_bottom_bright_f.getInterpolated(
+					sky_bottom_bright_night_f, color_change_fraction);
+
+				m_sky_color_top_bright_f = m_sky_color_top_bright_f.getInterpolated(
+					sky_top_bright_night_f, color_change_fraction);
 			} else {  // Day
-				m_bgcolor_bright_f = m_bgcolor_bright_f.getInterpolated(
-					bgcolor_bright_normal_f, color_change_fraction);
-				m_skycolor_bright_f = m_skycolor_bright_f.getInterpolated(
-					skycolor_bright_normal_f, color_change_fraction);
+				m_sky_color_bottom_bright_f = m_sky_color_bottom_bright_f.getInterpolated(
+					sky_bottom_bright_normal_f, color_change_fraction);
+
+				m_sky_color_top_bright_f = m_sky_color_top_bright_f.getInterpolated(
+					sky_top_bright_normal_f, color_change_fraction);
 			}
 
 			m_cloudcolor_bright_f = m_cloudcolor_bright_f.getInterpolated(
 				cloudcolor_bright_normal_f, color_change_fraction);
 		}
 	} else {
-		m_bgcolor_bright_f = m_bgcolor_bright_f.getInterpolated(
-			bgcolor_bright_indoor_f, color_change_fraction);
-		m_skycolor_bright_f = m_skycolor_bright_f.getInterpolated(
-			bgcolor_bright_indoor_f, color_change_fraction);
+		m_sky_color_bottom_bright_f = m_sky_color_bottom_bright_f.getInterpolated(
+			sky_bottom_bright_indoor_f, color_change_fraction);
+		m_sky_color_top_bright_f = m_sky_color_top_bright_f.getInterpolated(
+			sky_bottom_bright_indoor_f, color_change_fraction);
 		m_cloudcolor_bright_f = m_cloudcolor_bright_f.getInterpolated(
 			cloudcolor_bright_normal_f, color_change_fraction);
 		m_clouds_visible = false;
 	}
```

Comment:
The entire code block above has unnecessary changes due to an unnecessary changing of variable names.

Reply:
Actually, since in MT master we have
`video::SColor cloudyfogcolor = m_bgcolor;`
It seems clearest to change `bgcolor` to `cloudyfogcolor` everywhere. I agree with you that `bgcolor` is unclear and confusing. But please keep `skycolor` unchanged as it is accurate and familiar. So then the names are improved but changes are minimised.